MiniBootcamp/Composition/MainCoordinator.swift

```diff
@@ -19,7 +19,13 @@ class MainCoordinator: Coordinator {
     }
     
     func start() {
-        rootViewController.pushViewController(viewControllerFactory.feedViewController(), animated: false)
+        let feedVC = viewControllerFactory.feedViewController() as! FeedViewController
+        feedVC.coordinator = self
+        rootViewController.pushViewController(feedVC, animated: false)
```

Comment:
One of the advantages of Factory Patterns is to provide exactly and only what you need. So I would make the `feedViewController` method, to actually return a `FeedViewController` type. Or make it receive the coordinator and return a simple reusable `UIViewController` without the need of casting